File: ws/alban/u2/ws14ProductPrice/StoreProducts.G.U.I/src/storeproducts/g/u/i/Product.java
package storeproducts.g.u.i;

public class Product {
    private final String name;
    private final double price;
    private final int quantity;

    public Product(String name, double price, int quantity) {
        this.name = name;
        this.price = price;
        this.quantity = quantity;
    }

    public String getName() { return name; }
    public double getPrice() { return price; }
    public int getQuantity() { return quantity; }
    public double getSubtotal() { return this.price * this.quantity; }

    public String toJsonString() {
        return String.format("{\"name\": \"%s\", \"price\": %.2f, \"quantity\": %d, \"subtotal\": %.2f}", 
                name, price, quantity, getSubtotal());
    }
}

File: hw/lomas/u2/hw14DBMongoandGUI/hw14DBMongoandGUI.js/model/product.js
class Product {
    constructor(productId, name, price) {
        this.productId = productId;
        this.name = name;
        this.price = price;
    }

    toDict() {
        return {
            _id: this.productId,
            name: this.name,
            price: this.price
        };
    }
}

module.exports = Product;